### DISCUSSION TITLE
Podman Desktop reports that the Virtual Machine Platform is not enabled, but it is already enabled.

### DISCUSSION THREAD

**Discussion by zikzakjack** [podman-desktop/podman-desktop#7413] [Q&A]
Hello Community,

After uninstalling the older version, I am trying to set up Podman Desktop and Podman on my Windows Laptop. I keep getting the error below on the check requirements screen. Could someone help me overcome this issue?

`Virtual Machine Platform should be enabled to be able to run Podman
`
![image](https://github.com/containers/podman-desktop/assets/15928270/b504a63b-53ce-410a-aa4f-b202292d8a74)

### Step 1 - Enable the Windows Subsystem for Linux

```
C:\WINDOWS\system32>dism.exe /online /enable-feature /featurename:Microsoft-Windows-Subsystem-Linux /all /norestart

Deployment Image Servicing and Management tool
Version: 10.0.19041.844

Image Version: 10.0.19045.3208

Enabling feature(s)
[==========================100.0%==========================]
The operation completed successfully.

```

### Step 3 - Enable Virtual Machine feature

```
C:\WINDOWS\system32>dism.exe /online /enable-feature /featurename:VirtualMachinePlatform /all /norestart

Deployment Image Servicing and Management tool
Version: 10.0.19041.844

Image Version: 10.0.19045.3208

Enabling feature(s)
[==========================100.0%==========================]
The operation completed successfully.

```

![image](https://github.com/containers/podman-desktop/assets/15928270/0dc4d66d-c04f-47b3-853c-2b363be1fa81)

### Step 5 - Set WSL 2 as your default version

```
C:\WINDOWS\system32>wsl --set-default-version 2
For information on key differences with WSL 2 please visit https://aka.ms/wsl2
The operation completed successfully.

```

**Comment by odockal** [comment]
@zikzakjack Hey, thanks for opening the discussion. I see that you have enabled all you need in order to start using podman/podman desktop on your system. Only thing that come to my mind, have you also restarted the system after enabling the features?

**Reply by zikzakjack** [reply]
@odockal 
Hello There, Thanks for taking the time to look into this. 

1) Well, yes, I restarted at every step when it was required. I uninstalled the Podman CLI and Desktop and repeated all the steps. But the issue persisted. This happened only when I tried using the Podman Desktop to check the prerequisites and install the Podman CLI further.

2) However, I could successfully install the Podman CLI manually using the msi. After the manual installation, CLI was detectable within the Desktop where I was able to do the init/start, etc. Refer to below screenshot.

It is still unclear why the Podman Desktop was unable to detect the Virtual Machine Platform which was already enabled...

![image](https://github.com/containers/podman-desktop/assets/15928270/d0bd64db-3a39-465c-bf61-fe14c9b8508e)

**Reply by odockal** [reply]
@zikzakjack That is actually good workaround. Podman Desktop does [this](https://github.com/containers/podman-desktop/blob/main/extensions/podman/src/podman-install.ts#L612) to verify if the VMP was installed, probably you could try to reproduce the command from your powershell to see why it failed. What comes to my mind, could it be that you have enable it as an Admin and now you runs as another user, perhaps? Please try the command, if you still want to figure it out.

thanks!

**Reply by odockal** [reply]
You need to adjust the command a bit: `(Get-WmiObject -Query "Select * from Win32_OptionalFeature where InstallState = '1'").Name | select-string VirtualMachinePlatform`. Successful execution should return something like this:
```

VirtualMachinePlatform

```

**Reply by lukegeor** [reply]
I get the following when I try to run that query from powershell.exe:
```
PS C:\Users\me> Get-WmiObject -Query "Select * from Win32_OptionalFeature where InstallState = '1'"
Get-WmiObject : Invalid class "Win32_OptionalFeature"
At line:1 char:1
+ Get-WmiObject -Query "Select * from Win32_OptionalFeature where Insta ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidType: (:) [Get-WmiObject], ManagementException
    + FullyQualifiedErrorId : GetWMIManagementException,Microsoft.PowerShell.Commands.GetWmiObjectCommand
```

**Reply by odockal** [reply]
@lukegeor This seems weird. It looks like WMI services are not running. I have found something similar here: https://github.com/SpotX-Official/SpotX/issues/376.

**Reply by ThiagoFurlan32** [reply]
Hello,
@odockal @lukegeor 
I believe this is due to the fact that `Get-WmiObject` is not available in the latest versions of powershell and the correct option would be: `Get-CimInstance`.
I have the same problem, but when I run it with: `(Get-CimInstance -Query "Select * from Win32_OptionalFeature where InstallState = '1'").Name | Select-String VirtualMachinePlatform`. Returns: `VirtualMachinePlatform` normally.

https://github.com/PowerShell/PowerShell/issues/4766

**Reply by zikzakjack** [reply]
Hello @odockal @lukegeor @ThiagoFurlan32

I am unable to explore further on this issue as I happened to return the laptop back to the client. Anyway. Thank you all for the time...

**Reply by bodzebod** [reply]
Hi, I have the same problem on my work laptop (windows 11, powershell 7.4.5).
I also saw the same problem reported in issues 7414 and 7211.
When I run the VM feature detection command in powershell, it returns it:
![image](https://github.com/user-attachments/assets/4cccfbbe-5c29-4a98-97d1-ae404294b9d4)

But podman keeps on not detecting it:
![image](https://github.com/user-attachments/assets/2c56ce79-312c-458a-a34f-ed100701f00a)

I uninstalled and restarted the installer a few times. To be noted that I also get such error when trying to setup compose:
![image](https://github.com/user-attachments/assets/aa558532-c520-4417-a1c4-f0841a91885d)
Despite the following mention, I never get a prompt for asking for system-wide privileges:
![image](https://github.com/user-attachments/assets/e6b40f3f-bf9e-4a03-90fa-bc0493aacc52)

As a workaround, I have installed podman manually by grabbing the latest release on https://github.com/containers/podman/releases . Then podman desktop detected it and proposed to start it.
In addition, I downloaded manually the last docker-compose EXE from https://github.com/docker/compose/releases and put it in podman binaries folder c:\program files\RedHat\podman, then podman desktop was able to complete the compose setup.

**Reply by odockal** [reply]
@ThiagoFurlan32 Thanks for your findings! 
@bodzebod I have opened an issue so we can adjust the call and make sure any host's powershell system can handle the installation: https://github.com/containers/podman-desktop/issues/9539

**Comment by dgolovin** [comment]
I think this is encoding problem similar to https://github.com/containers/podman-desktop/pull/4918, because as doc for powershell says [here](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_character_encoding?view=powershell-7.4#character-encoding-in-windows-powershell) default encoding is utf16le.

**Comment by Straven** [comment]
Had the same problem and installing podman manually solves the problem.

**Reply by LeonardoLeite** [reply]
Just install Podman trough "winget", and the issue disappears.

**Comment by yesabhishek** [comment]
yep, had the same problem. Downloaded the `podman` from the [Releases](https://github.com/containers/podman/releases) fixed it.

**Comment by Rob-El** [comment]
I downloaded podman desktop 1.25.1, encountered the same problem (Powershell 7.5.4).  Tried uninstalling desktop and then installing desktop via winget.  Same issue.  

Then I realized there's a podman desktop, and a separate podman CLI (5.7.1), that's what Desktop is trying to install.  Ohhhhh, desktop download site doesn't mention CLI.  That's not obvious from the downloads page, but if it works, then why would it?

"winget install RedHat.Podman" worked, even though the desktop still complained about WSL.  Desktop is slow to refresh and took a couple restarts before it recognized that podman cli was installed.  Created my first pod/container, off to the races!

**Comment by Inspirateur** [comment]
Why is this a discussion ? This is clearly an issue. I'm also running into it

**Comment by odockal** [comment]
@Inspirateur can you please open an issue with steps to reproduce? thx
